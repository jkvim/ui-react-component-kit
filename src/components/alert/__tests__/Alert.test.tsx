import * as React from 'react';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Alert } from '../';

describe('Alert', () => {

  describe('props', () => {

    const wrapper = mount(
      <Alert type="success" className="class">Hello</Alert>);

    it('should match snapshot', () => {
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    it('should pass `type` in props', () => {
      expect(wrapper.props().type).toBe('success');
    });

    it('should pass `className` in props', () => {
      expect(wrapper.props().className).toBe('class');
    });

    it('should pass `children` in props', () => {
      expect(wrapper.props().children).toBe('Hello');
    });
  });

  it('snapshot basic', () => {
    const wrapper = mount(<Alert className="class">Hello</Alert>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot success', () => {
    const wrapper = mount(<Alert type="success" className="class">Hello</Alert>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot info', () => {
    const wrapper = mount(<Alert type="info" className="class">Hello</Alert>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot warning', () => {
    const wrapper = mount(<Alert type="warning" className="class">Hello</Alert>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('snapshot danger', () => {
    const wrapper = mount(<Alert type="danger" className="class">Hello</Alert>);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});