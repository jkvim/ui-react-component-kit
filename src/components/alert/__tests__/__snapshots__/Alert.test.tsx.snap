// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert props should match snapshot 1`] = `
<Alert
  className="class"
  type="success"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="success"
  >
    <div
      className="class sc-bdVaJa OSvWk"
      role="alert"
      type="success"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert snapshot basic 1`] = `
<Alert
  className="class"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
  >
    <div
      className="class sc-bdVaJa jVmMKh"
      role="alert"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert snapshot danger 1`] = `
<Alert
  className="class"
  type="danger"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="danger"
  >
    <div
      className="class sc-bdVaJa iooWOT"
      role="alert"
      type="danger"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert snapshot info 1`] = `
<Alert
  className="class"
  type="info"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="info"
  >
    <div
      className="class sc-bdVaJa eZexUJ"
      role="alert"
      type="info"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert snapshot success 1`] = `
<Alert
  className="class"
  type="success"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="success"
  >
    <div
      className="class sc-bdVaJa OSvWk"
      role="alert"
      type="success"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;

exports[`Alert snapshot warning 1`] = `
<Alert
  className="class"
  type="warning"
>
  <styled.div
    className="class"
    role="alert"
    tabindex="-1"
    theme={
      Object {
        "border": Object {
          "radius": Object {
            "edge": "2px",
            "round": "100px",
          },
          "size": "2px",
        },
        "breakpoint": Object {
          "lg": "992px",
          "md": "768px",
          "sm": "576px",
          "xl": "1200px",
          "xs": "0px",
        },
        "color": Object {
          "background": "#ffffff",
          "border": "#2fa4e7",
          "checked": "#2fa4e7",
          "danger": "#c71c22",
          "default": "#2fa4e7",
          "disabled": "#acacac",
          "error": "#c71c22",
          "focus": "#54b4eb",
          "font": "#333333",
          "hover": "#178acc",
          "info": "#04519b",
          "primary": "#2fa4e7",
          "secondary": "#e9ecef",
          "success": "#73a839",
          "tertiary": "#343a40",
          "warning": "#dd5600",
        },
        "font": Object {
          "size": Object {
            "default": "1.8rem",
          },
          "weight": Object {
            "default": 100,
          },
        },
      }
    }
    type="warning"
  >
    <div
      className="class sc-bdVaJa djmiuG"
      role="alert"
      type="warning"
    >
      Hello
    </div>
  </styled.div>
</Alert>
`;
